core/types/wo.go

```diff
@@ -318,18 +319,20 @@ func (wo *WorkObject) InterlinkHashes() common.Hashes {
 }
 
 func (wo *WorkObject) QiTransactions() []*Transaction {
-	return wo.Body().QiTransactions()
-}
-
-func (wo *WorkObject) QuaiTransactions() []*Transaction {
-	return wo.Body().QuaiTransactions()
+	qiTxs := make([]*Transaction, 0)
+	for _, t := range wo.Transactions() {
+		if t.Type() == QiTxType {
+			qiTxs = append(qiTxs, t)
+		}
+	}
+	return qiTxs
 }
 
 func (wo *WorkObject) QiTransactionsWithoutCoinbase() []*Transaction {
 	// TODO: cache the UTXO loop
 	qiTxs := make([]*Transaction, 0)
-	for i, t := range wo.Transactions() {
-		if i == 0 && IsCoinBaseTx(t, wo.woHeader.parentHash, wo.woHeader.location) {
+	for _, t := range wo.Transactions() {
+		if IsCoinBaseTx(t) {
```

Comment:
You can actually delete this, because this function ignores ETXs altogether

Reply:
deleted